packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -332,6 +332,74 @@ abstract class ImageProvider<T> {
     return ImageStream();
   }
 
+  /// Checks if this [ImageProvider] has a matching key in the
+  /// [PaintingBinding.imageCache]
+  ///
+  /// The `cache` and `configuration` parameters must not be null. If the
+  /// `handleError` parameter is null, errors will be reported to
+  /// [FlutterError.onError], and the method will return null.
+  ///
+  /// A completed return value of null indicates that an error has occurred.
+  Future<ImageCacheLocation> findCacheLocation({
+    @required ImageConfiguration configuration,
+    ImageErrorListener handleError,
+  }) {
+    assert(configuration != null);
+    final Completer<ImageCacheLocation> completer = Completer<ImageCacheLocation>();
+    bool didError = false;
+    final ImageErrorListener errorHandler = (dynamic exception, StackTrace stack) {
+      if (didError) {
+        return;
+      }
+      didError = true;
+      if (handleError != null) {
+        handleError(exception, stack);
+      } else {
+        FlutterError.onError(FlutterErrorDetails(
+          context: ErrorDescription('while checking the cache location of an image'),
+          informationCollector: () sync* {
+            yield DiagnosticsProperty<ImageProvider>('Image provider', this);
+            yield DiagnosticsProperty<ImageConfiguration>('Image configuration', configuration);
+          },
+          exception: exception,
+          stack: stack,
+        ));
+        completer.complete(null);
+      }
+    };
+
+
+    // If an error is added to a synchronous completer before a listener has been
```

Comment:
This pice of code looks familiar. Any change we can factor it out in some kind of a common place so we only have to do this hackery in one place?

Reply:
I'll talk offline - I'm worried that if we do that, someone will use an implementation import to get at it and it try to use it, which we don't want.

Reply:
Wait, isn't the other occurence also in this file? Could this be just be refactored into a private method? I don't think there's any danger of anybody accessing a private method from here?